include/deal.II/sundials/ida.h

```diff
@@ -0,0 +1,653 @@
+     * -  none: do not try to make initial conditions consistent.
+     * -  use_y_diff: compute the algebraic components of y and differential
+     *    components of y_dot, given the differential components of y.
+     *    This option requires that the user specifies differential and
+     *    algebraic components in the function get_differential_components.
+     * -  use_y_dot: compute all components of y, given y_dot.
+     *
+     * Notice that you could in principle use this capabilities to solve for
+     * stady state problems by setting y_dot to zero, and asking to compute
+     * $y(0)$ that satisfies $F(0, y(0), 0) = 0$, however the nonlinear solver
+     * used inside IDA may not be robust enough for complex problems with
+     * several millions unknowns.
+     */
+    std::string ic_type;
+
+    /**
+     * Type of conditions to be used after a solver restart.
+     *
+     * If you do not have consistent initial conditions after a restart, (i.e.,
+     * conditions for which F(y_dot(t_restart), y(t_restart), t_restart) = 0),
+     * you can ask SUNDIALS to compute the new initial conditions for you by
+     * using the `reset_type` parameter at construction time.
+     *
+     * You have three options
+     * -  none: do not try to make initial conditions consistent.
+     * -  use_y_diff: compute the algebraic components of y and differential
+     *    components of y_dot, given the differential components of y.
+     *    This option requires that the user specifies differential and
+     *    algebraic components in the function get_differential_components.
+     * -  use_y_dot: compute all components of y, given y_dot.
+     */
+    std::string reset_type;
+
+    /**
+     * Alpha to use in Newton method for IC calculation.
+     */
+    double ic_alpha;
+
+    /**
+     * Maximum number of iterations for Newton method in IC calculation.
+     */
+    unsigned ic_max_iter;
+
+    /**
+     * Maximum number of iterations for Newton method during time advancement.
+     */
+    unsigned int max_non_linear_iterations;
+
+    /**
+     * Show the progress of the time stepper.
+     */
+    bool verbose;
+
+    /**
+     * Use local tolerances when computing absolute tolerance.
+     */
+    bool use_local_tolerances;
+
+    /**
+     * IDA memory object.
+     */
+    void *ida_mem;
+
+    /**
+     * IDA solution vector.
+     */
+    N_Vector yy;
+
+    /**
+     * IDA solution derivative vector.
+     */
+    N_Vector yp;
+
+    /**
+     * IDA absolute tolerances vector.
+     */
+    N_Vector abs_tolls;
+
+    /**
+     * IDA differential components vector.
+     */
+    N_Vector diff_id;
+
+#ifdef DEAL_II_WITH_MPI
+    /**
+     * MPI communicator. SUNDIALS solver runs happily in parallel.
+     */
+    MPI_Comm communicator;
+#endif
+
+    /**
+     * Output stream. If run in parallel, only processor zero will
+     * output verbose information about time steps.
+     */
+    ConditionalOStream pcout;
```

Comment:
@dealii/dealii if I understand correctly we are moving away form writing anything from dealii's classes and instead provide functions for optional output